data-prepper-plugins/http-source/src/main/java/org/opensearch/dataprepper/plugins/source/loghttp/LogHTTPService.java

```diff
@@ -87,9 +87,10 @@ private HttpResponse processRequest(final AggregatedHttpRequest aggregatedHttpRe
         }
         try {
             if (buffer.isByteBuffer()) {
-                // jsonList is ignored in this path but parse() was done to make 
-                // sure that the data is in the expected json format
-                buffer.writeBytes(content.array(), null, bufferWriteTimeoutInMillis);
+                for (final String json: jsonList) {
+                    final String jsonArray = "["+json+"]";
```

Comment:
Let's use Jackson to make this JSON array rather than rely on string building. We have similar string manipulation in our code and it may introduce errors.

You can use the streaming API of Jackson or ObjectMapper.